store/sqlstore/link_metadata_store.go

```diff
@@ -47,33 +47,33 @@ func (s SqlLinkMetadataStore) Save(metadata *model.LinkMetadata) (*model.LinkMet
 
 	err := s.GetMaster().Insert(metadata)
 	if err != nil && !IsUniqueConstraintError(err, []string{"PRIMARY", "linkmetadata_pkey"}) {
-		return nil, model.NewAppError("SqlLinkMetadataStore.Save", "store.sql_link_metadata.save.app_error", nil, "url="+metadata.URL+", "+err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "could not save link metadata")
 	}
 
 	return metadata, nil
 }
 
-func (s SqlLinkMetadataStore) Get(url string, timestamp int64) (*model.LinkMetadata, *model.AppError) {
+func (s SqlLinkMetadataStore) Get(url string, timestamp int64) (*model.LinkMetadata, error) {
 	var metadata *model.LinkMetadata
 	query, args, err := s.getQueryBuilder().
 		Select("*").
 		From("LinkMetadata").
 		Where(sq.Eq{"URL": url, "Timestamp": timestamp}).
 		ToSql()
 	if err != nil {
-		return nil, model.NewAppError("SqlLinkMetadataStore.Get", "store.sql.build_query.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "could not create query with querybuilder")
 	}
 	err = s.GetReplica().SelectOne(&metadata, query, args...)
 	if err != nil {
 		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlLinkMetadataStore.Get", "store.sql_link_metadata.get.app_error", nil, "url="+url+", "+err.Error(), http.StatusNotFound)
+			return nil, store.NewErrNotFound("LinkMetadata", "url="+url)
 		}
-		return nil, model.NewAppError("SqlLinkMetadataStore.Get", "store.sql_link_metadata.get.app_error", nil, "url="+url+", "+err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "could not get metadata with url=%s", url)
```

Comment:
For SQL query errors, let's just prefix with the query method, in this case, "selectone". So maybe something like "selectone: url=%s"

Reply:
Done